### PR TITLE
Unmute, fix, refactor and zen2ify NetworkDisruptionIT

### PR DESCRIPTION
**Pull request by andrershov**
This PR unmutes `NetworkDisruptionIT` on master.

It makes changes necessary for Zen2 - avoids usage of autoMinMasterNodes and selects cluster size, such that there is no need to call `AddVotingExclusion`.
This test also introduces refactors a single method `prepareDistruptedCluster` to be used by both test methods.
Unfortunately, `NetworkDisruption` is broken and the testNetworkPartitionRemovalRestoresConnections "is fixed" by introducing `assertBusy` -  https://github.com/elastic/elasticsearch/issues/38348.
I want to avoid backporting this PR to 6.x branch because there are might be Zen1 specific issues with the tests as well.

Relates https://github.com/elastic/elasticsearch/issues/36205
Relates https://github.com/elastic/elasticsearch/issues/38348

### CODE REVIEW

File: test/framework/src/test/java/org/elasticsearch/test/disruption/NetworkDisruptionIT.java [hunk 1]
```diff
@@ -35,52 +44,82 @@
...
+        assertThat(nodes.size(), greaterThanOrEqualTo(3));
+        int majority = nodes.size() / 2 + 1;
+        Set<String> side1 = new HashSet<>(randomSubsetOf(randomIntBetween(majority, nodes.size() - 1), nodes));
+        assertThat(side1.size(), greaterThanOrEqualTo(majority));
         Set<String> side2 = new HashSet<>(nodes);
         side2.removeAll(side1);
         assertThat(side2.size(), greaterThanOrEqualTo(1));
         NetworkDisruption networkDisruption = new NetworkDisruption(new TwoPartitions(side1, side2),
-            new NetworkDisruption.NetworkDisconnect());
+                new NetworkDisruption.NetworkUnresponsive());
```
Comment (ywelsch) [inline test/framework/src/test/java/org/elasticsearch/test/disruption/NetworkDisruptionIT.java]: why change this to unresponsive?

**Review by ywelsch — CHANGES_REQUESTED** [review]
Is the assertBusy and lowering `NodeConnectionsService.CLUSTER_NODE_RECONNECT_INTERVAL_SETTING` not sufficient here?
Note that these settings from AbstractDisruptionTestCase have their own problems, as they won't allow the cluster to be stable on a slow CI worker.

**Comment by andrershov** [comment]
@ywelsch thank you for your review, I've made necessary changes. Can you please make the second pass?

**Review by ywelsch — APPROVED** [review]
LGTM

**Pull request merged**